source/common/router/scoped_rds.cc

```diff
@@ -19,6 +19,57 @@ using Envoy::Config::ConfigProviderPtr;
 namespace Envoy {
 namespace Router {
 
+namespace ScopedRoutesConfigProviderUtil {
+
+ConfigProviderPtr
+create(const envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager&
+           config,
+       Server::Configuration::FactoryContext& factory_context, const std::string& stat_prefix,
+       ConfigProviderManager& scoped_routes_config_provider_manager) {
+  ASSERT(config.route_specifier_case() == envoy::config::filter::network::http_connection_manager::
+                                              v2::HttpConnectionManager::kScopedRoutes);
+
+  switch (config.scoped_routes().config_specifier_case()) {
+  case envoy::config::filter::network::http_connection_manager::v2::ScopedRoutes::
+      kScopedRouteConfigurationsList: {
+    const envoy::config::filter::network::http_connection_manager::v2::
+        ScopedRouteConfigurationsList& scoped_route_list =
+            config.scoped_routes().scoped_route_configurations_list();
+    ProtobufTypes::ConstMessagePtrVector config_protos;
+    std::transform(scoped_route_list.scoped_route_configurations().begin(),
+                   scoped_route_list.scoped_route_configurations().end(),
+                   std::back_inserter(config_protos),
+                   [](const envoy::api::v2::ScopedRouteConfiguration& scoped_route_config)
+                       -> std::unique_ptr<const Protobuf::Message> {
+                     Protobuf::Message* clone = scoped_route_config.New();
+                     clone->MergeFrom(scoped_route_config);
+                     return std::unique_ptr<const Protobuf::Message>(clone);
+                   });
+    return scoped_routes_config_provider_manager.createStaticConfigProvider(
+        std::move(config_protos), factory_context,
+        ScopedRoutesConfigProviderManagerOptArg(config.scoped_routes().name(),
+                                                config.scoped_routes().rds_config_source(),
+                                                config.scoped_routes().scope_key_builder()));
+  }
+
+  case envoy::config::filter::network::http_connection_manager::v2::ScopedRoutes::kScopedRds:
+    return scoped_routes_config_provider_manager.createXdsConfigProvider(
+        config.scoped_routes().scoped_rds(), factory_context, stat_prefix,
+        ScopedRoutesConfigProviderManagerOptArg(config.scoped_routes().name(),
+                                                config.scoped_routes().rds_config_source(),
+                                                config.scoped_routes().scope_key_builder()));
+
+  case envoy::config::filter::network::http_connection_manager::v2::ScopedRoutes::
+      CONFIG_SPECIFIER_NOT_SET:
```

Comment:
We do have one, see [http_connection_manager.proto](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L516-L531).

I've removed handling of the condition to allow it to fall through to the `default:` to avoid ambiguity.